client/signup/config/steps-pure.js

```diff
@@ -576,6 +576,23 @@ export function generateSteps( {
 			providesDependencies: [ 'bearer_token', 'email', 'username' ],
 			unstorableDependencies: [ 'bearer_token' ],
 		},
+		'prelaunch-domains': {
+			stepName: 'prelaunch-domains',
+			apiRequestFunction: createSiteWithCart,
+			providesDependencies: [
+				'siteId',
+				'siteSlug',
+				'domainItem',
+				'themeItem',
+				'shouldHideFreePlan',
+			],
+			optionalDependencies: [ 'shouldHideFreePlan' ],
```

Comment:
I added this line with mirrors `domains` step:

https://github.com/Automattic/wp-calypso/blob/144ac119b6d89c743e8790b6a5563311c07dd421/client/signup/config/steps-pure.js#L250-L257

It seems to fix the issue I was encountering.